### PR TITLE
use guzzlehttp/guzzle instead of deprecated guzzle/guzzle

### PR DESCRIPTION
**Pull request by OskarStark**
I am targetting this branch, because this is a BC change.

## Changelog

```markdown
### Changed
- dependency from `guzzle/guzzle` to `guzzlehttp/guzzle`, because it is deprecated
```

## Subject

Package guzzle/guzzle is abandoned, you should avoid using it. Use guzzlehttp/guzzle instead.

Old one: https://packagist.org/packages/guzzle/guzzle
New one: https://packagist.org/packages/guzzlehttp/guzzle

### CODE REVIEW

**Comment by OskarStark** [comment]
Lets see if travis is green

**Comment by Soullivaneuh** [comment]
AFAIK, I think guzzle 3.* under the new namespace was a mistake and not all the release are in this place. So we should not use it.

**Comment by OskarStark** [comment]
i compared 3.8.0 release on both github repos and the have the same commit hash

**Comment by OskarStark** [comment]
ping @greg0ire

**Comment by greg0ire** [comment]
LGTM, but I'm a bit worried about people being unable to upgrade if they already use, say, `guzzlehttp/guzzle` 4, 5 or 6 in their project :S

**Comment by greg0ire** [comment]
Maybe this should be considered major…

**Comment by OskarStark** [comment]
> LGTM, but I'm a bit worried about people being unable to upgrade if they already use, say, guzzlehttp/guzzle 4, 5 or 6 in their project :S

What you are saying sounds reasonable to me, shall I provide a PR against `master` and add an UPGRADE note?

**Comment by core23** [comment]
> Maybe this should be considered major…

What about switching to http://httplug.io/ with the next major?

File: composer.json [hunk 1]
```diff
@@ -35,13 +35,13 @@
     "suggest": {
         "sonata-project/admin-bundle": "To be able to use MessageAdmin.",
         "sonata-project/doctrine-orm-admin-bundle": "^3.0",
-        "guzzle/guzzle" : "^3.9",
+        "guzzlehttp/guzzle" : "^3.9",
```
Comment (core23) [inline composer.json]: Can you update the description to text instead of a wrong version notation.

Comment (OskarStark) [inline composer.json]: Which text should i add?

**Comment by OskarStark** [comment]
> What about switching to http://httplug.io/ with the next major?

sounds good

File: composer.json (hunk 1 above)
Comment (core23) [inline composer.json]: Be creative ;)

Comment (greg0ire) [inline composer.json]: "You won't believe what happens when requiring this package!"

Comment (core23) [inline composer.json]: These are the only lines with Guzzle usage:
https://github.com/sonata-project/SonataNotificationBundle/blob/85bd86a34e6139f1b92d45be6769c539eb8e7366/Backend/AMQPBackendDispatcher.php#L200-L209

**Comment by greg0ire** [comment]
> What you are saying sounds reasonable to me, shall I provide a PR against master and add an UPGRADE note?

Oh wait! You are touching the `suggest` section, not the `require` section, so who cares ? It has no impact, this is good to merge IMO.

**Comment by OskarStark** [comment]
> Oh wait! You are touching the suggest section

and `require-dev`

**Comment by greg0ire** [comment]
> and require-dev

Oh right, but it's the same : no one cares, you can change it.

File: composer.json (hunk 1 above)
Comment (OskarStark) [inline composer.json]: maybe we remove it completely from the suggest section?

Comment (greg0ire) [inline composer.json]: Nah. Just put : 

> if you want to get a status report when using the rabbitMQbackend

or something like that.

Comment (OskarStark) [inline composer.json]: Thanks, done!

**Comment by greg0ire** [comment]
Can you apply [this comment](https://github.com/sonata-project/SonataNotificationBundle/pull/210#issuecomment-243802734)? It's the only thing left I think.

**Comment by OskarStark** [comment]
> Can you apply this comment? It's the only thing left I think.

done and formatted a few exception messages

**Comment by greg0ire** [comment]
Aaaaaand StyleCI fails. Same player shoot again :gun:

**Comment by OskarStark** [comment]
done 👍

File: Backend/AMQPBackendDispatcher.php [hunk 2]
```diff
@@ -53,7 +53,7 @@ public function __construct(array $settings, array $queues, $defaultQueue, array
     }
 
     /**
-     * @return AMQPChannelannel
+     * @return AMQPChannel
```
Comment (greg0ire) [inline Backend/AMQPBackendDispatcher.php]: lol good catch! Thank god there are people who actually read the comments :)

**Comment by greg0ire** [comment]
LGTM

**Comment by core23** [comment]
We need a BC fix for this:
https://github.com/sonata-project/SonataNotificationBundle/blob/85bd86a34e6139f1b92d45be6769c539eb8e7366/Backend/AMQPBackendDispatcher.php#L200-L209

**Comment by greg0ire** [comment]
> We need a BC fix for this:
https://github.com/sonata-project/SonataNotificationBundle/blob/85bd86a34e6139f1b92d45be6769c539eb8e7366/Backend/AMQPBackendDispatcher.php#L200-L209

Are you sure of this? The package name change does not imply a namespace change IIRC. Only the error message needs to be changed IMO.

**Comment by greg0ire** [comment]
> Only the error message needs to be changed IMO.

Which was done. LGTM

**Comment by greg0ire** [comment]
@core23 : I think as long we use 3.x there is no BC-break

**Comment by core23** [comment]
> Are you sure of this? The package name change does not imply a namespace change IIRC. Only the error message needs to be changed IMO.

You're right. The namespace changed in Guzzle 4.0

**Comment by greg0ire** [comment]
Perfect! I'll let you merge then.

**Comment by core23** [comment]
Thank you @OskarStark

**Pull request merged**